AutoCollection/Statsbeat.ts

```diff
@@ -148,15 +164,52 @@ class Statsbeat {
         this._retryCount++;
     }
 
+    public async trackNetworkStatsbeatMetrics() {
+        this._getResourceProvider(async () => {
+            let networkProperties = {
+                "os": this._os,
+                "rp": this._resourceProvider,
+                "cikey": this._cikey,
+                "runtimeVersion": this._runtimeVersion,
+                "language": this._language,
+                "version": this._sdkVersion,
+                "attach": this._attach,
+                "instrumentation": this._instrumentations,
+            }
+            this._trackRequestDuration(networkProperties);
+            this._trackRequestsCount(networkProperties);
+            await this._sendStatsbeats();
+        });
+    }
+
     public async trackStatsbeatMetrics() {
         this._getResourceProvider(async () => {
-            this._trackRequestDuration();
-            this._trackRequestsCount();
+            let attachProperties = {
```

Comment:
@heyams the spec doesn't mention attach property to determine if "codeless" or "sdk" for this Statsbeat can you confirm if is by design?

Reply:
attach statsbeats is different from attach type, which is codeless or sdk.  attach statsbeat is used to collect rp and rpid for attach rates later. hope that makes sense.  i could rename attach in the customer dimensions to sdkType? it's either codeless or sdk.

Reply:
You can see this exactly what the code is supposed to be doing here
https://github.com/microsoft/ApplicationInsights-E2E-Tests/blob/main/AppInsightsE2ETest/E2ETestCases/TestsConfiguration.cs#L90
It must only use setting value when env var is not present, it would be great if you can fix the issue in the code we are all using in this case